services/authService.js

```diff
@@ -21,13 +23,43 @@ export const getUserJWT = () => {
   return Promise.resolve(userInfoCache.jwt);
 }
 
+export const pollUserInfo = (integrationUrl, token) => {
+    var attempts = 0;
+    var promise = new Promise(function(resolve) {
+        var interval = setInterval(() => {
+            getUserSession(integrationUrl, token, false)
+            .then(() => {
+                // 200 - OK, return to the previous flow
+                clearInterval(interval);
+                resolve({
+                  success: success,
+                  jwt: token,
+                });
+            })
+            .catch((error) => {
+                const response = error.response || {};
+                if (response.status !== 401) {
+                    clearInterval(interval);
+                    throw error;
+                }
+                // 401, keep polling
+                if (++attempts >= MAX_NUM_OF_ATTEMPTS) {
+                    clearInterval(interval);
+                    resolve(false);
```

Comment:
This could be resolved using the same template as the others: `{success: false}`